Binary files xorg-server-ubuntu-git-patched/.git/index and xorg-server-ubuntu-git-working/.git/index differ
diff -Nurp xorg-server-ubuntu-git-patched/os/log.c xorg-server-ubuntu-git-working/os/log.c
--- xorg-server-ubuntu-git-patched/os/log.c	2009-12-01 18:49:34.000000000 -0800
+++ xorg-server-ubuntu-git-working/os/log.c	2009-12-01 20:29:19.000000000 -0800
@@ -314,6 +314,29 @@ LogVMessageVerb(MessageType type, int ve
     const char *s  = X_UNKNOWN_STRING;
     char tmpBuf[1024];
 
+    struct timeval time;
+    time_t tv_sec;
+    suseconds_t tv_usec;
+    static Bool first = TRUE;
+    static time_t start_tv_sec;
+    static suseconds_t start_usec;
+    int diff_sec, diff_usec;
+
+    gettimeofday(&time, NULL);
+    tv_sec = time.tv_sec;
+    tv_usec = time.tv_usec;
+    if (first == TRUE) {
+        start_tv_sec = tv_sec;
+        start_usec = tv_usec;
+        first = FALSE;
+    }
+    diff_sec = (int)difftime(tv_sec, start_tv_sec);
+    diff_usec = (tv_usec - start_usec);
+    if (diff_usec < 0) {
+        diff_sec--;
+        diff_usec += 1000000;
+    }
+
     /* Ignore verbosity for X_ERROR */
     if (logVerbosity >= verb || logFileVerbosity >= verb || type == X_ERROR) {
 	switch (type) {
@@ -355,7 +378,8 @@ LogVMessageVerb(MessageType type, int ve
 	}
 
         /* if s is not NULL we need a space before format */
-        snprintf(tmpBuf, sizeof(tmpBuf), "%s%s%s", s ? s : "",
+        snprintf(tmpBuf, sizeof(tmpBuf), "[%5d.%06d] %s%s%s", diff_sec, diff_usec,
+                                                   s ? s : "",
                                                    s ? " " : "",
                                                    format);
         LogVWrite(verb, tmpBuf, args);
